### PR TITLE
add fixed-length lists

### PR DESCRIPTION
**Pull request by aspizu**

### CODE REVIEW

File: src/codegen/sb3.rs [hunk 1]
```diff
@@ -938,6 +938,16 @@ where T: Write + Seek
                     vec![]
                 }
             },
+            Some(ListDefault::FixedLength(value, length)) => {
+                let value = s.evaluate_const_expr(d, value);
+                let len = s.evaluate_const_expr(d, length).to_number();
```
Comment (aspizu) [inline src/codegen/sb3.rs:949]: **MEDIUM** — A negative or fractional length passes this upper-bound-only check, and the later `as usize` cast silently changes it: `[0; -1]` becomes empty and `[0; 2.9]` gets length 2. That makes a fixed-length declaration compile to a different size instead of showing a source error. Check that the length is finite, non-negative, and whole before casting, then add tests for a negative and a fractional length.

File: src/codegen/sb3.rs [hunk 2]
```diff
@@ -938,6 +938,16 @@ where T: Write + Seek
...
+                if len > 200_000_f64 {
+                    d.report(DiagnosticKind::ListTooBig, &length.span());
+                    vec![]
+                } else {
+                    vec![value; len as usize]
```
Comment (aspizu) [inline src/codegen/sb3.rs]: **MEDIUM** — This makes only `len` flat cells for every list type, but a struct list is stored as one Scratch list per field and the code below splits `data` with `data.len() / struct_.fields.len()`. A two-field struct declared with `[0; 10]` therefore gets only 5 items per field (and short lengths can become empty). Handle fixed-length struct lists by making one value per field for each requested item, or reject this form for struct lists, and add a two-field regression test.

**Review by aspizu — COMMENTED** [review]
Code review: two medium-severity findings.
